x-pack/plugins/security_solution/cypress/integration/alerts_detection_rules_custom.spec.ts

```diff
@@ -104,23 +106,23 @@ import { loginAndWaitForPageWithoutDateRange } from '../tasks/login';
 
 import { DETECTIONS_URL } from '../urls/navigation';
 
-const expectedUrls = newRule.referenceUrls.join('');
-const expectedFalsePositives = newRule.falsePositivesExamples.join('');
-const expectedTags = newRule.tags.join('');
-const expectedMitre = formatMitreAttackDescription(newRule.mitre);
-const expectedNumberOfRules = 1;
-const expectedEditedtags = editedRule.tags.join('');
-const expectedEditedIndexPatterns =
-  editedRule.index && editedRule.index.length ? editedRule.index : indexPatterns;
-
-describe('Custom detection rules creation', () => {
-  before(() => {
-    esArchiverLoad('timeline');
+describe('Custom detection rules creation', async () => {
+  const expectedUrls = newRule.referenceUrls.join('');
+  const expectedFalsePositives = newRule.falsePositivesExamples.join('');
+  const expectedTags = newRule.tags.join('');
+  const expectedMitre = formatMitreAttackDescription(newRule.mitre);
+  const expectedNumberOfRules = 1;
+
+  let rule: typeof newRule;
+
+  before(async () => {
+    const createdTimeline = await createTimeline(newRule.timeline);
+    rule = deepMerge(newRule, { timeline: { id: createdTimeline[0] } });
```

Comment:
Nit: I don't think `deepmerge` is strictly necessary, here:

```ts
rule = { ...newRule, { timeline: { ...newRule.timeline, id: timelineId } } }
```
I would go that route for now and keep dependencies to a minimum.